src/ReportGenerator.Core/Parser/DynamicCodeCoverageParser.cs

```diff
@@ -83,7 +83,7 @@ public ParserResult Parse(XContainer report)
         /// <returns>The <see cref="Assembly"/>.</returns>
         private Assembly ProcessAssembly(XElement module)
         {
-            string assemblyName = module.Attribute("name").Value;
+            string assemblyName = module.Attribute("name").Value + "_" + module.Attribute("id").Value;
```

Comment:
This will prevent the tool from mixing code coverage data from different versions of the same module (same name) together. E.g. x86 and x64 version of the same DLL.

Reply:
Are you generating the XML file by hand? I.e. are you manually copying the relevant section from the x86 coverage file to the x64 file or vice versa?
If yes: How about renaming the modules to "???_x86" and "???_x64" if you want separate results? The id looks "ugly" in the report.
In generell users would expect that modules with the same name are handled as a single module. I'm not willing to change that behavior, because users rely on it. Often ReportGenerator is used to merge several coverage files.
You are the first user that does not want to merge the results :-)